aspnetcore/web-api/index.md

```diff
@@ -101,15 +131,25 @@ namespace WebApiSample
 
 ## Attribute routing requirement
 
-The `ApiController` attribute makes attribute routing a requirement. For example:
+The `[ApiController]` attribute makes attribute routing a requirement. For example:
+
+::: moniker range=">= aspnetcore-3.0"
 
-[!code-csharp[](index/samples/2.x/Controllers/ValuesController.cs?name=snippet_Signature&highlight=1)]
+[!code-csharp[](index/samples/3.x/Controllers/WeatherForecastController.cs?name=snippet_ControllerSignature&highlight=2)]
+
+::: moniker-end
+
+::: moniker range="<= aspnetcore-2.2"
+
+[!code-csharp[](index/samples/2.x/Controllers/ValuesController.cs?name=snippet_ControllerSignature&highlight=1)]
+
+::: moniker-end
 
 Actions are inaccessible via [conventional routes](xref:mvc/controllers/routing#conventional-routing) defined by <xref:Microsoft.AspNetCore.Builder.MvcApplicationBuilderExtensions.UseMvc*> or <xref:Microsoft.AspNetCore.Builder.MvcApplicationBuilderExtensions.UseMvcWithDefaultRoute*> in `Startup.Configure`.
```

Comment:
@rynowak How does this statement change with the inclusion of endpoint routing in the 3.0 web API project template?

Reply:
@rynowak How does this statement change with the inclusion of endpoint routing in the 3.0 web API project template?
@dougbu  who's the backup for @rynowak  on endpoint routing?

Reply:
@rynowak is around. @JamesNK is his backup.

Reply:
@JamesNK How does this statement change with the inclusion of endpoint routing in the 3.0 web API project template?

Reply:
It shouldn't have changed. Endpoint routing is designed to be backwards compatible as much as possible with the older routing API.

Reply:
@JamesNK My question wasn't specific enough. Should the 3.0 version of this text say that actions are inaccessible via conventional routes defined by the following?

```csharp
            app.UseEndpoints(endpoints =>
            {
                endpoints.MapControllerRoute(
                    name: "default",
                    pattern: "{controller=Home}/{action=Index}/{id?}");
            });
```

Reply:
Yes, the same concept applies. Conventional routes in UseMvc and conventional routes in UseEndpoints are both overridden by attribute routing.